### ISSUE TITLE
TFLint does not fail when an aws_instance inside a module have invalid type

### ISSUE THREAD

**Issue by IslamAzab** [terraform-linters/tflint#502]
```hcl
// main.tf
module "test_module" {
  source = "./modules/test-module"
}

// modules/test-module/main.tf
resource "aws_instance" "foo" {
  ami           = "ami-0ff8a91507f77f867"
  instance_type = "t1.2xlarge" # invalid type!
}
```

## Version

Terraform v0.12.6
+ provider.aws v2.21.1
+ provider.google v2.12.0
+ provider.template v2.1.2

## Expected Behaviour

TFLint should fail when an **`aws_instance` inside a module** have invalid type

**Comment by IslamAzab** [comment]
Thanks for getting back so quickly. 

I can do the recursive check of all directories too.

I have 2 questions:
- So does this mean that **[Module Inspection](https://github.com/wata727/tflint/blob/v0.12.1/docs/guides/advanced.md#module-inspection)**, will check variables if it only has the same names as resources?
- will you enable this recursive check in `TFLint` in the future?

**Comment by wata727** [comment]
> So does this mean that Module Inspection, will check variables if it only has the same names as resources?

No, this means checking the module **arguments**.  See the example below:

```hcl
# main.tf
module "ec2" {
  source = "./module_path"
  type = "t1.2xlarge"
}
```

```hcl
# module_path/main.tf
variable "type" {}

resource "aws_instance" "foo" {
  ami = "ami-123456"
  instance_type = var.type # invalid type
}

resource "aws_instance" "bar" {
  ami = "ami-123456"
  instance_type = "t1.2xlarge" # no issues found
}
```

Since the `bar` resource's issue is irrelevant to the argument of the `ec2` module, no issue will be reported when inspecting the root module. This is not related to variable names or resource names.

> will you enable this recursive check in TFLint in the future?

This is a very difficult problem. The answer is No at the moment.

Intuitively, I understand that Linter's behavior requires recursive checking. However, there is no recursive plan/apply in Terraform, so the behavior that TFLint should simulate is undefined.

For example, `path.cwd` depends on the current directory. In addition, the behavior changes depending on environment variables such as `TF_DATA_DIR` and `TF_WORKSPACE`. These variables can change depending on the directory you are running, and must be determined explicitly.

We may be able to avoid this complexity by treating only the modules specially, but I have no ideas yet.

**Comment by wata727** [comment]
Close this due to no activity.
